File: 2023/01_Segmentation/Python/functionReadTIFFMultipage.py
import cv2
import numpy as np
from PIL import Image

def functionReadTIFFMultipage(dirImage, bitdepth):
    img = Image.open(dirImage)
    images = []
    for i in range(img.n_frames):
        img.seek(i)
        images.append(np.array(img))

    del img

    height, width = np.shape(images[0])
    numImgs = len(images)
    #print(height, width, numImgs)

    if bitdepth == 8:
        volume = np.uint8(np.zeros((height, width, numImgs)))
    else:
        volume = np.uint16(np.zeros((height, width, numImgs)))

    for i in range(numImgs):
        sliceSingle = images[i]
        volume[:, :, i] = sliceSingle
    
    return volume

File: 2023/01_Segmentation/MATLAB/aux_functions/functionReadTIFFMultipage.m
function [volume,fileInfo] = functionReadTIFFMultipage(dirImage)

fileInfo = imfinfo(dirImage);
numImgs = size(fileInfo,1);
volume = uint16(zeros(fileInfo(1).Height,fileInfo(1).Width,numImgs));

for i=1:numImgs%For each tile
   slice = imread(dirImage,i, 'Info', fileInfo);
   volume(:,:,i) = slice;
end

end

